src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,11 +1,40 @@
 package core.basesyntax.service;
 
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    public static final int MIN_AGE = 18;
+    public static final int MIN_PASSWORD_LENGTH = 6;
+    private final StorageDao storage;
+
+    public RegistrationServiceImpl() {
+        this.storage = new StorageDaoImpl();
+    }
+
+    private void validateUser(User user) {
+        if (user == null
+                || user.getLogin() == null
+                || user.getPassword() == null
+                || user.getAge() == null
+        ) {
+            throw new RuntimeException("Some user data or user may be null");
+        }
+    }
 
     @Override
     public User register(User user) {
-        return null;
+        validateUser(user);
+        if (user.getId() == null
+                && storage.get(user.getLogin()) == null
```

Comment:
I think this check really better move to the separate 'if' with a specific message. Because it is not about data requirements